realm-transformer/src/main/groovy/io/realm/transformer/BytecodeModifier.groovy

```diff
@@ -147,4 +149,15 @@ class BytecodeModifier {
                     new CtClass[0], new CtClass[0], "{return true;}", clazz))
         }
     }
+
+    private static CtMethod createSetter(String methodName, CtField field) {
+        CtMethod setter = CtNewMethod.setter(methodName, field)
+        if ("io.realm.RealmInteger".equals(field.getType().getName())) {
+            setter.insertBefore(
+                    'if (!($1 instanceof io.realm.internal.datatypes.realminteger.UnmanagedRealmInteger))' +
+                    'throw new IllegalStateException("Cannot assign a managed RealmInteger to an unmanaged object");');
```

Comment:
This setter is only for unmanaged object, isn't it?

If so, I don't think that we need to reject `UnmanagedRealmInteger`.

At least all existing setters are only for unmanaged objects since those are overridden by setters defined in the corresponding proxy class.